### PR TITLE
End to end encryption

### PR DESCRIPTION
**Pull request by griffobeid**

### CODE REVIEW

File: yew-ui/src/components/attendants.rs [hunk 1]
```diff
@@ -65,6 +74,8 @@ pub struct AttendantsComponentProps {
     #[prop_or_default]
     pub email: String,
 
+    pub e2ee_enabled: bool,
```
Comment (darioalessandro) [inline yew-ui/src/components/attendants.rs:77]: This is awesome, in a follow up MR we can enable/disable encryption via a URL parameter

File: yew-ui/src/components/attendants.rs [hunk 2]
```diff
@@ -208,14 +255,88 @@ impl Component for AttendantsComponent {
...
             },
             Msg::OnInboundMedia(response) => {
-                if let Err(e) = self.peer_decode_manager.decode(response) {
-                    log!("error decoding packet: {:?}", e);
+                match response.packet_type.enum_value() {
+                    Ok(PacketType::AES_KEY) => {
+                        if !self.e2ee_enabled {
+                            return false;
+                        }
+                        log!("Received AES_KEY", &response.email);
```
Comment (darioalessandro) [inline yew-ui/src/components/attendants.rs:275]: Is there a way to set a log level like on the backend? this is great for debugging and I do want to keep it around, I would just like to hide it in prod

File: yew-ui/src/components/attendants.rs [hunk 3]
```diff
@@ -208,14 +255,88 @@ impl Component for AttendantsComponent {
...
+                            );
+                        }
+                        return false;
+                    }
+                    Ok(PacketType::RSA_PUB_KEY) => {
+                        if !self.e2ee_enabled {
+                            return false;
+                        }
+                        log!("Received RSA_PUB_KEY");
+                        let rsa_packet = RsaPacket::parse_from_bytes(&response.data).unwrap();
```
Comment (darioalessandro) [inline yew-ui/src/components/attendants.rs:294]: these unwraps might cause an issue down the line, is there a way to use `and_then` and log the errors if any?

File: yew-ui/src/components/attendants.rs [hunk 4]
```diff
@@ -208,14 +255,88 @@ impl Component for AttendantsComponent {
...
             Msg::OnInboundMedia(response) => {
-                if let Err(e) = self.peer_decode_manager.decode(response) {
-                    log!("error decoding packet: {:?}", e);
+                match response.packet_type.enum_value() {
+                    Ok(PacketType::AES_KEY) => {
+                        if !self.e2ee_enabled {
+                            return false;
+                        }
+                        log!("Received AES_KEY", &response.email);
+                        if let Ok(bytes) = self.rsa.decrypt(&response.data) {
```
Comment (darioalessandro) [inline yew-ui/src/components/attendants.rs:276]: What happens when the ui fails to decrypt the message?

**Review by darioalessandro — COMMENTED** [review]
This is awesome!!

File: yew-ui/src/components/attendants.rs (hunk 2 above)
Comment (griffobeid) [reply yew-ui/src/components/attendants.rs:275]: Yes I can update this MR with that

**Review by darioalessandro — APPROVED** [review]
lgtm!!!

**Pull request merged**